docs: add graceful error boundary example (#77781)
vercel/next.js: docs/01-app/04-api-reference/03-file-conventions/error.mdx

}
```

> ### Graceful Error Recovery with a Custom Error Boundary

Reviewer: ```suggestion
### Graceful error recovery with a custom error boundary
```

> When rendering fails on the client, it’s often useful to show the last known good UI than a blank screen for a better user experience.

Reviewer 2: ```suggestion
When rendering fails on the client, it can be useful to show the last known server rendered UI for a better user experience.
```
